### PR TITLE
Tuner: handle the scenario when the trial is stopped externally, i.e. independently of the Scheduler

### PR DESCRIPTION
**Pull request by talesa**
*Description of changes:*
Hello!
This patch handles the scenario when the trial is stopped externally, i.e. independently of the Scheduler, e.g. due to max_run argument (the timeout setting of the SM Estimator) or manually via the AWS console.
I encountered this issue when my individual trials had `max_run=10min`, and when they were stopped (by themselves), the `running_trials_ids` in `Tuner.run()` was not updated and hence the Tuner wouldn't start more trials.

The proposed solution is the best I could do at first glance, but there might be a more elegant way to handle this.
This logic of the proposed solution should be carefully reviewed by someone with a good understanding of the Tuner class, because I'm not 100% certain if it's fully consistent with other logic of this class.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: syne_tune/tuner.py [hunk 1]
```diff
@@ -417,14 +428,18 @@ def _update_running_trials(
...
             if status == Status.failed:
                 logger.info(f"Trial trial_id {trial_id} failed.")
                 self.scheduler.on_trial_error(trial)
                 done_trials[trial_id] = (trial, status)
 
+            # For the case when the trial is stopped independently of the scheduler, we choose to use
+            # scheduler.on_trial_error(...) since it was not the scheduler's decision to stop the trial.
+            if status == Status.stopped and trial_id not in self.trials_scheduler_stopped:
+                logger.info(f"Trial trial_id {trial_id} was stopped independently of the scheduler.")
+                self.scheduler.on_trial_error(trial)
```
[the remark this answers is not in the document]
Comment (geoalgo) [reply syne_tune/tuner.py:440]: I think this is fine for now to call any external interruption an error, we can revisit this later.

**Review by geoalgo — COMMENTED** [review]
LGTM, thanks for doing this change I have been wanting to do it for a while. 
I have only two small remarks on comments, otherwise the logic looks good to me (another thing to is to have mockup to test the tuner logic against such edge cases).

File: syne_tune/tuner.py [hunk 2]
```diff
@@ -364,7 +369,9 @@ def _update_running_trials(
         Trials can be finished because:
          1) the scheduler decided to stop or pause.
          2) the trial failed.
-         3) the trial completed.
+         3) the trial was stopped independently of the scheduler, e.g. due to max_run argument, the timeout setting of
```
[the remark this answers is not in the document]
Comment (talesa) [reply syne_tune/tuner.py]: Addressed

**Review by geoalgo — APPROVED** [review]

**Comment by talesa** [comment]
> another thing to is to have mockup to test the tuner logic against such edge cases

@geoalgo do we have an existing mockup test of this kind I could adapt for this case?

**Comment by geoalgo** [comment]
> @geoalgo do we have an existing mockup test of this kind I could adapt for this case?

No we dont, we would benefit from having it but its a significant effort it does not have to be done in this PR.
